### PR TITLE
fix(net): reject response errors in QGETDATA requests

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented

`CQuorumDataRequest` reads an optional trailing `nError` byte for QDATA responses, but the QGETDATA request handler also accepted that response-only field. A requester could select one of the missing-data errors that intentionally suppress repeated-request scoring while still forcing response work.

This is the protocol-validation half split out of #7519 so it can be reviewed and merged independently from request-tracking bounds.

## What was done?

- Reject QGETDATA payloads longer than the fixed request encoding with a score of 100.
- Detect the field by presence rather than decoded value, so explicit `NONE`, `UNDEFINED`, and trailing bytes are rejected too.
- Extend the functional-test message helper to construct the malformed request and verify that the peer is disconnected.

## How Has This Been Tested?

- Full local `make -j13` build on macOS arm64 using the prebuilt depends tree.
- `/opt/homebrew/bin/python3.9 test/functional/test_runner.py p2p_quorum_data.py`
- `git diff --check`

`test/lint/lint-python.py` was invoked but skipped because `flake8` is not installed in the local environment.

## Breaking Changes

None. `nError` on an inbound QGETDATA request was never meaningful.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone

This pull request was created by Codex.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Final review complete — no blockers (commit 1a32e0e7)

File: test/functional/p2p_quorum_data.py [hunk 1]
```diff
@@ -376,6 +376,19 @@ def send_bad_qdata_expect_disconnect(bad_qdata):
...
+        def test_qgetdata_rejects_requester_error():
+            self.log.info("Test QGETDATA with requester-supplied nError is disconnected")
+            p2p_mn = p2p_connection(mn2.get_node(self))
+            id_p2p_mn = get_p2p_id(mn2.get_node(self))
+            mnauth(mn2.get_node(self), id_p2p_mn, fake_mnauth_2[0], fake_mnauth_2[1])
+            wait_for_banscore(mn2.get_node(self), id_p2p_mn, 0)
+            poisoned = msg_qgetdata(quorum_hash_int, 100, 0x01, error=ENCRYPTED_CONTRIBUTIONS_MISSING)
+            p2p_mn.send_message(poisoned)
+            self.wait_until(lambda: not p2p_mn.is_connected, timeout=10)
+            mn2.get_node(self).disconnect_p2ps()
```
Comment (thepastaclaw) [inline test/functional/p2p_quorum_data.py]: **🟡 Suggestion: Exercise presence-based rejection for both special error bytes**

The PR explicitly requires rejection based on field presence, including serialized `NONE` (`0x00`) and `UNDEFINED` (`0xFF`), but the test sends only `ENCRYPTED_CONTRIBUTIONS_MISSING`. It would therefore pass with a weaker value-based check that rejects ordinary nonzero errors while accepting one or both special values. Exercise both boundary values on fresh peers, while retaining the existing nonzero case, so the regression test distinguishes the size-based implementation from every decoded-value gate.

```suggestion
            for error in (0, 0xFF, ENCRYPTED_CONTRIBUTIONS_MISSING):
                p2p_mn = p2p_connection(mn2.get_node(self))
                id_p2p_mn = get_p2p_id(mn2.get_node(self))
                mnauth(mn2.get_node(self), id_p2p_mn, fake_mnauth_2[0], fake_mnauth_2[1])
                wait_for_banscore(mn2.get_node(self), id_p2p_mn, 0)
                poisoned = msg_qgetdata(quorum_hash_int, 100, 0x01, error=error)
                p2p_mn.send_message(poisoned)
                self.wait_until(lambda: not p2p_mn.is_connected, timeout=10)
                mn2.get_node(self).disconnect_p2ps()
```

<sub>source: ['codex', 'coderabbit']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Final validation — Codex/Sol only (Phase 2 disabled)

The fixed-size payload gate correctly rejects every response-only or trailing byte before deserialization and applies the intended misbehavior score. The functional test covers the original nonzero-error exploit, but it does not verify the presence-based behavior promised for the two special decoded values.
Source: reviewer backend model: gpt-5.6-sol; final verifier backend model: gpt-5.6-sol. Orchestration-only: openclaw-agent/cliproxy/gpt-5.6-sol (not reviewer evidence).

Validated zero-blocker Codex/Sol precheck evidence was promoted to final because Phase 2 (Sonnet/Opus) is temporarily disabled. This is Codex/Sol-only final validation, not Codex + Sonnet/Opus coverage.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed)
- Verifier: `gpt-5.6-sol` — verifier
- Sonnet/Opus: **not run (Phase 2 disabled — temporary Codex/Sol-only final)**
- Secondary pass: disabled (`temporary_phase2_sonnet_disable`)

🟡 1 suggestion(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `test/functional/p2p_quorum_data.py`:
- [SUGGESTION] test/functional/p2p_quorum_data.py:383-390: Exercise presence-based rejection for both special error bytes
  The PR explicitly requires rejection based on field presence, including serialized `NONE` (`0x00`) and `UNDEFINED` (`0xFF`), but the test sends only `ENCRYPTED_CONTRIBUTIONS_MISSING`. It would therefore pass with a weaker value-based check that rejects ordinary nonzero errors while accepting one or both special values. Exercise both boundary values on fresh peers, while retaining the existing nonzero case, so the regression test distinguishes the size-based implementation from every decoded-value gate.
```
</details>

**Review by thepastaclaw — APPROVED** [review]
## Final validation — Codex/Sol only (Phase 2 disabled)

The fixed-size QGETDATA payload check rejects response-only error fields and arbitrary trailing bytes before deserialization. The current functional test exercises the ordinary missing-data error plus explicit NONE (0x00) and UNDEFINED (0xFF) values on fresh connections, so the prior coverage finding is fixed and no in-scope issues remain.
Source: reviewer backend model: gpt-5.6-sol; final verifier backend model: gpt-5.6-sol. Orchestration-only: openclaw-agent/cliproxy/gpt-5.6-sol (not reviewer evidence).

Validated zero-blocker Codex/Sol precheck evidence was promoted to final because Phase 2 (Sonnet/Opus) is temporarily disabled. This is Codex/Sol-only final validation, not Codex + Sonnet/Opus coverage.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed)
- Verifier: `gpt-5.6-sol` — verifier
- Sonnet/Opus: **not run (Phase 2 disabled — temporary Codex/Sol-only final)**
- Secondary pass: disabled (`temporary_phase2_sonnet_disable`)

File: test/functional/p2p_quorum_data.py (hunk 1 above)
Comment (thepastaclaw) [reply test/functional/p2p_quorum_data.py]: Resolved in `1a32e0e` — _Exercise presence-based rejection for both special error bytes_ no longer present.

<sub>Auto-resolved by the review system based on the latest commit diff. If you believe this was closed in error, reopen the thread.</sub>

File: test/functional/p2p_quorum_data.py [hunk 2]
```diff
@@ -376,6 +376,19 @@ def send_bad_qdata_expect_disconnect(bad_qdata):
...
+
```
[the remark this answers is not in the document]
Comment (thepastaclaw) [reply test/functional/p2p_quorum_data.py:391]: This is no longer applicable at the exact head. Commit 1a32e0e7fff updated lines 382-390 to test ENCRYPTED_CONTRIBUTIONS_MISSING, explicit NONE (0x00), and explicit UNDEFINED (0xFF), using a fresh authenticated peer and disconnect assertion for each value.
